README.md

```diff
@@ -67,18 +67,79 @@ $ DOCKERFILE=myfile TAG=mytag make image
 ## Configuration Reference
 
 User can provide the following configurations as environment variables or for the ConfigMap :
-* `DAEMON_SM_PLUGIN`: Name of the subnet manager plugin. Currently supported `"noop"` and `"ufm"`.
+* `DAEMON_SM_PLUGIN`: Name of the subnet manager plugin.
 * `DAEMON_PERIODIC_UPDATE`: Interval in seconds to send add and remove request to subnet manager.
 * `GUID_POOL_RANGE_START`: The first guid in the pool e.g: `"02:00:00:00:00:00:00:00"`.
 * `GUID_POOL_RANGE_END`: The Last guid in the pool.
 
-**Configurations if "ufm" subnet manager plugin is used for  `deployment/ib-kubernetes-ufm-secret.yaml`:**
-* `UFM_USERNAME`: Username of UFM.
-* `UFM_PASSWORD`: Password of UFM.
-* `UFM_ADDRESS`: IP address or hostname of UFM server.
-* `UFM_HTTP_SCHEMA`: http/https, default is https.
-* `UFM_PORT`: REST API port of UFM default is 443 (https), if `httpSchema` is set to http then the default is 80.
-* `UFM_CERTIFICATE`: Secure certificate if using secure connection.
+## Plugins
+
+Subnet Manager Plugin to configure PKeys (Partition Key) in the InfiniBand fabric.
+
+### NOOP Plugin
+
+Plugin that does nothing.
+
+### UFM (Unified Fabric Manager) Plugin
+
+[UFM](https://www.mellanox.com/products/management-software/ufm) is a powerful platform for managing scale-out computing environments.
+UFM Plugin allow to configure PKeys (Partition Key) via UFM.
+
+#### Plugin Configration
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: ib-kubernetes-ufm-secret
+  namespace: kube-system
+stringData:
+  UFM_USERNAME: "admin"  # UFM Username
+  UFM_PASSWORD: "123456" # UFM Password
+  UFM_ADDRESS: ""        # UFM Hostname/IP Address 
+  UFM_HTTP_SCHEMA: ""    # http/https. Default: httpd
+  UFM_PORT: ""           # UFM REST API port: https default 443 and http default 80
+string:
+  UFM_CERTIFICATE: ""    # UFM Certificate in bas64 format . (if not provided client crtificate verfication is skipt)
+```
+
+#### Self Signed Certificates
+
+login to UFM
+
+UFM on container
+``` 
+$ docker exec -it ufm bash
+```
+
+create private key and certificate
+```
+$ openssl req -x509 -newkey rsa:4096 -keyout ufm.key -out ufm.crt -days 365 -subj '/CN=<UFM hostname>'
+```
+
+install UFM private key and certificate
+```
+$ cp ufm.key /etc/pki/tls/certs/ufmlocalhost.key
+$ cp ufm.crt /etc/pki/tls/private/ufmlocalhost.crt
+
+```
+Restart UFM 
+
+UFM on container
+```
+$ docker restart ufm
+```
+
+Create UFM secret
+```
+$ kubectl create secret generic ib-kubernetes-ufm-secret --namespace="kube-system" --from-literal=UFM_USER="admin" --from-literal=UFM_PASSWORD="12345" --from-literal=UFM_ADDRESS="127.0.01" --from-file=UFM_CERTIFICATE=ufmlocalhost.crt --dry-run -o yaml > ib-kubernetes-ufm-secret.yaml
+$ kubectl create -f ./ib-kubernetes-ufm-secret.yaml 
+```
+
+UFM on bare-metal
```

Comment:
why does ufm on container restart before secret creation and ufm on baremetal after ?

can we have both restarts then the secret creation ?

i.e
UFM restart
  containerized: ...
  baremetal: ....
Create secret ....